handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -865,14 +865,19 @@ public final SSLEngineResult wrap(
                         bytesWritten = writePlaintextData(src, min(remaining, availableCapacityForWrap));
                     }
 
+                    // Determine how much encrypted data was generated.
+                    //
+                    // Even if SSL_write did fail we may produced some data in the dst buffer (as the return value
```

Comment:
Consider rewording to make the statement more general and use similar terminology as used elsewhere:

> Even if `SSL_write` doesn't consume any application data it is possible that OpenSSL will produce non-application data into the BIO. For example session tickets....

Reply:
are session tickets the only scenario where this might occur?

Reply:
I think in theory no...
It was the only way how to reproduce this for me tho